runtime/parser/declaration_test.go

```diff
@@ -2557,6 +2557,150 @@ func TestParseEvent(t *testing.T) {
+													Type: &ast.NominalType{
+														Identifier: ast.Identifier{
+															Identifier: "String",
+															Pos: ast.Position{
+																Offset: 43,
+																Line:   1,
+																Column: 43,
+															},
+														},
+													},
+													StartPos: ast.Position{
+														Offset: 43,
+														Line:   1,
+														Column: 43,
+													},
+												},
+												DefaultArgument: &ast.StringExpression{
+													Value: "foo",
+													Range: ast.Range{
+														StartPos: ast.Position{
+															Offset: 52,
+															Line:   1,
+															Column: 52,
+														},
+														EndPos: ast.Position{
+															Offset: 56,
+															Line:   1,
+															Column: 56,
+														},
+													},
+												},
+												Identifier: ast.Identifier{
+													Identifier: "a",
+													Pos: ast.Position{
+														Offset: 39,
+														Line:   1,
+														Column: 39,
+													},
+												},
+												StartPos: ast.Position{
+													Offset: 39,
+													Line:   1,
+													Column: 39,
+												},
+											},
+										},
+										Range: ast.Range{
+											StartPos: ast.Position{
+												Offset: 37,
+												Line:   1,
+												Column: 37,
+											},
+											EndPos: ast.Position{
+												Offset: 57,
+												Line:   1,
+												Column: 57,
+											},
+										},
+									},
+									StartPos: ast.Position{
+										Offset: 37,
+										Line:   1,
+										Column: 37,
+									},
+									Access: ast.AccessNotSpecified,
+								},
+								Kind: common.DeclarationKindInitializer,
+							},
+						},
+					),
+					Identifier: ast.Identifier{
+						Identifier: "ResourceDestroyed",
+						Pos: ast.Position{
+							Offset: 19,
+							Line:   1,
+							Column: 19,
+						},
+					},
+					Range: ast.Range{
+						StartPos: ast.Position{
+							Offset: 1,
+							Line:   1,
+							Column: 1,
+						},
+						EndPos: ast.Position{
+							Offset: 57,
+							Line:   1,
+							Column: 57,
+						},
+					},
+					Access:        ast.AccessAll,
+					CompositeKind: common.CompositeKindEvent,
+				},
+			},
+			result,
+		)
+	})
+
+	t.Run("default event with no default arg", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, errs := testParseDeclarations(" access(all) event ResourceDestroyed ( a : Int )")
+
+		utils.AssertEqualWithDiff(t, []error{
+			&SyntaxError{
+				Pos:     ast.Position{Line: 1, Column: 47, Offset: 47},
+				Message: "expected a default argument after type annotation, got ')'",
+			},
+		}, errs)
+	})
+
+	t.Run("non-default event with default arg", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, errs := testParseDeclarations(" access(all) event Foo ( a : Int = 3)")
+
+		utils.AssertEqualWithDiff(t, []error{
+			&SyntaxError{
```

Comment:
Can you please also add a test for parsing a function expression and function declaration which have parameter lists with default arguments, even though that is unexpected?

Reply:
@dsainati1 Did a test already exist?

Reply:
I added them here in 15543eb9ab6665c8bcb9dd08d3d88bcb29c00897

Reply:
Ah, I missed that, thanks!